### PR TITLE
handle failed mounts

### LINKED ISSUE #50556: Iscsi disk manager should handle failed mounts
The iscsi disk manager should handle failed mounts correctly see [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/iscsi/disk_manager.go)

/sig storage
/assign

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #50556

**Special notes for your reviewer**:
/cc @yastij 

**Release note**:

```release-note
None
```

### CODE REVIEW

File: pkg/volume/fc/disk_manager.go [hunk 1]
```diff
@@ -36,12 +37,14 @@ type diskManager interface {
...
 func diskSetUp(manager diskManager, b fcDiskMounter, volPath string, mounter mount.Interface, fsGroup *int64) error {
 	globalPDPath := manager.MakeGlobalPDName(*b.fcDisk)
-	// TODO: handle failed mounts here.
 	noMnt, err := mounter.IsLikelyNotMountPoint(volPath)
-
-	if err != nil && !os.IsNotExist(err) {
-		glog.Errorf("cannot validate mountpoint: %s", volPath)
-		return err
+	if err != nil {
+		if !os.IsNotExist(err) {
```
Comment (yastij) [inline pkg/volume/fc/disk_manager.go]: Please PTAL to #50502  to see what I'm talking about.
These changes should be handled separatly, @saad-ali to you for confirmation

**Comment by yastij** [comment]
@dixudx - this doesn't fix #50556.

**Comment by dixudx** [comment]
@yastij Sorry for that. Closed this PR. In favor of your PRs.

**Pull request closed without merging**